### ISSUE TITLE
Secure boot

### ISSUE THREAD

**Issue by iulko** [r0b0/debian-installer#58]
Why secure boot cant be enabled with this installer when official debian installer can be?

**Comment by r0b0** [comment]
Hi,

No real reason, I just didn't care for it enough :) I've read the [debian secure boot page](https://wiki.debian.org/SecureBoot) and it all seemed an awful lot of work.

I will try to find some time and implement this into the installer.

**Comment by r0b0** [comment]
The problem is that we are using systemd-boot instead of grub and debian refused to sign systemd-boot with their keys - see https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=996202

**Comment by r0b0** [comment]
Also, the debian shim-signed package depends on grub and does not support systemd-boot - see https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1039058

**Comment by flaviut** [comment]
Good news on https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=996202:

> From: Luca Boccassi <bluca@debian.org>
> To: 996202@bugs.debian.org
> Cc: waldi@debian.org, biebl@debian.org, debian-efi@lists.debian.org
> Subject: Re: Bug#996202: EFI Secure Boot for systemd-boot
> Date: Mon, 04 Mar 2024 02:13:25 +0000
> …
> The upstream Shim reviewers group now accepts systemd-boot as a 2nd stage bootloader, trusted by Shim builds signed with the UEFI 3rd party CA. This clears the way for Debian's CA to sign systemd-boot, so I am reopening this bug.

**Comment by TheDan64** [comment]
Is this able to progress now that the bugfix has been merged upstream? Or are there still other blockers?

**Comment by r0b0** [comment]
Can you test the trixie version, please?
https://github.com/r0b0/debian-installer/tree/testing
Systemd-boot is now signed and pulls shim automatically so it might just work. There is still [1086714](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1086714) which causes the grub package to be installed (not used though)

**Comment by r0b0** [comment]
> I've used ODIN in the past for a system with TPM but without secure boot. Is it possible to retroactively enable Secure boot now for this system and without reinstalling?

Yes, see here https://github.com/r0b0/debian-installer/wiki/Authenticated_Boot#instructions
